NOTICK: Upgrade to Corda Gradle plugins 7.0.2-SNAPSHOT.

chrisr3 opened the pull request.

nikinagy committed:
update versions

nikinagy committed:
CORE-9701 - Handle multiple versions of member info (#3055)

Ability to handle multiple versions of MemberInfo base don statuses. Added status based lookups and new keying strategies.
P2P layer got modified to use the status filters.

Commit:
CORE-11213: update to 700 to avoid collision (#3301)

Commit:
CORE-11196: dont swap branch for tag build on downstream jobs (#3284)

josephzunigadaly committed:
CORE-8774: DB connection failures must be recoverable - CpkWriteServiceImpl (#3119)

* Set status to DOWN when either `ConfigurationReadService` or `DbConnectionManager` move to DOWN or ERROR state
* Set status to UP when both `ConfigurationReadService` and `DbConnectionManager` move to UP state, schedule reconciliation task at this time
* Schedule reconciliation after setting status to UP in `onConfigChangedEvent` to make sure we only have the timer running while UP
* Cancel the scheduled reconciliation task in `closeResources`
* Keep `configReadServiceRegistration` open until stop event (not `closeResources`) so we can receive up/down notifications of `ConfigurationReadService` and `DbConnectionManager`
* Close `configSubscription` before creating a new one in `onRegistrationStatusChangeEvent`
* Use null safe operations in `scheduleNextReconciliationTask` so it is safe to call the method even before config is received
* Use getManagedResource for subscriptions

josephzunigadaly committed:
CORE-8774: DB connection failures must be recoverable - DBProcessorImpl (#3120)

Make `DBProcessorImpl` follow the lifecycle of `DbConnectionManager`.

Commit:
CORE-11213: ensure we have unique tags when building (#3307)

owenstanford committed:
CORE-7646 Added support in the virtual node writer service for creating nodes async (#3241)

CORE-7646 Added support in the virtual node writer service for creating nodes async

filipesoliveira committed:
CORE-8744 Align Cpi related entities and repository with latest JPA guidance (#3296)

* CORE-8744 Align Cpi related entities and repository with latest JPA guidance

Moved CpkDbChangeLogAuditEntity and CpkDbChangeLogEntity to an internal submodule so it is not exported by OSGi.

Fixed build issue regarding OSGi. Without this fix the error "error  : Export net.corda.libs.cpi.datamodel.repository,  has 1,  private references [net.corda.libs.cpi.datamodel.entities.internal]" is raised.

yift-r3 committed:
CORE-10347: Allow repersistent of the same group parameters (#3226)

* CORE-10347: Allow repersistent of the same group parameters

* Apply review comments

* Apply review comments

* Add kdoc

* mode docs

* Apply review comment

* detekt

mbrkic-r3 committed:
CORE-8983 NullPointerException when parking flow Fiber (#3282)

Shows unrecognized input command in RpcSmokeTestFlow that was previously failing with NPE. Fix was made in the Quasar library that was released with the same snapshot version.

relyafi committed:
CORE-8625 Update JPA backing store entities to follow best practice (#3270)

relyafi committed:
CORE-11181 Notary code updates to conform to engineering code guidelines (#3300)

filipesoliveira committed:
CORE-8744 Align Cpi related entities and repository with latest JPA guidance (#3244)

File checksums and cpi signer summary hashes are now strongly typed. An instance
of SecureHash is now used instead of String since the former provides more context to people.

CpiIdentifier type should be used in the methods instead of separate arguments.
This helps keeping the list of arguments for a method small.

Deleted PersistenceUtils.kt to simplify the code base. To help reduce complexity of the code,
variables should always have the same name across the code base and not variations of the
original name. If someone requires a string version of a SecureHash, they can call the method toString().

Other but small refactors were also made.

kyriathar-r3 committed:
CORE-11445 Log membership persistence request type in `MembershipPersistenceRPCProcessor` (#3115)

* Logs received membership persistence request type

YashNabar committed:
Merge remote-tracking branch 'origin/release/network-team-ga-scope' into yash/network-feat-merge

Commit:
CORE-11409: Rename string constants `rpc-api` and `rpc-api-tls` (#3313)

vlajos committed:
CORE-11440 Phase out MemberX500Name from some e2e-test-utilities arguments (#3322)

CORE-11440 Phase out MemberX500Name from some e2e-test-utilities arguments to let us remove corda api dependencies from the e2e-tests repo.

josephzunigadaly committed:
CORE-6971: Allow CordaSerializable for custom serializer target classes (#3298)

- Update `isCustomSerializationForbidden` to allow custom serializers when the target class is `CordaSerializable`
- Name the return values `forbidden` and `allowed` to improve readability
- Update unit test to reflect new behaviour
- Expand integration test to prove we can't target platform `CordaSerializable` types with custom serialisers
- Add `CordaSerializable` to target types of internal custom serializers

vlajos committed:
CORE-8638 Move getCpiMetadata, getCpkMetadata, getNumberOfComponentGroups, getSchemaVersion from TransactionMetadata to impl (#3320)

Tom-Fitzpatrick committed:
CORE-11405: Forward merge Beta2 branch into release/os/5.0 (#3325)

* CORE-11405: Make getVirtualNodeOperationStatus return 404 if no vNode Operation status found (#3315)

yift-r3 committed:
CORE-11410: Fix MGM plugin and onboard script (#3316)

* CORE-11410: Fix MGM plugin and onboaed script

* Apply review comment

YashNabar committed:
Merge remote-tracking branch 'origin/release/os/5.0' into yash/network-feat-merge

YashNabar committed:
CORE-11449 Merge Network team feature branch

owenstanford committed:
CORE-11464 Update timeout while waiting for test vNode creation. (#3329)

owenstanford committed:
CORE-11433 Remove a temporary timeout value for create virtual node now the endpoint has been updated to follow the async patten. (#3321)

kyriathar-r3 committed:
CORE-9678 Hibernate errors during crypto worker startup (#3312)

Following error was being observed at crypto worker start up:

ERROR: duplicate key value violates unique constraint "crypto_hsm_category_association_uc" Detail: Key (tenant_id, category, deprecated_at)=(rpc-api, JWT_KEY, 0) already exists. Failed to assign SOFT HSM for rpc-api:JWT_KEY 

This must have been caused due to DB find and persist calls not being transactional when it comes to `HSMStoreImpl.associate`.

* Makes `HSMStoreImpl.associate` DB find and persist calls transactional

owenstanford committed:
CORE-11441 Cleanup - Remove SecureHash utils from chunking-core  (#3323)

CORE-11441 Move SecureHash utils from chunking-core to crypto-core

YashNabar committed:
CORE-11449 Revert feature branch specific changes

josephzunigadaly committed:
CORE-8774: DB connection failures must be recoverable - DbReconcilerReader (#3121)

* Don't close DbReconcilerReader.dependencyRegistration, we need it to get the UP signal when DB comes back
* Migrate to managed resources API
* Delete tests checking registration handle is closed because DbReconcilerReader now delegates that to managed resources, so it is no longer a responsibility of this class

YashNabar committed:
CORE-11449 Corda-api version bump

davidcurrie committed:
CORE-8896 Add corda-lib chart (#3297)

YashNabar committed:
CORE-11449 Merge Network team feature branch

Merge temporary feature branch of network team back into main branch after beta 2 branch is cut.

conal-smith committed:
CORE-11292 liquibase diff exception handling (#3302)

CORE-11292 - Handle liquibase exception when triggering a virtual node upgrade
- if the CPI has faulty / corrupt XML, this exception leaves the virtual node with a non-null operationInProgress, leaving the vnode unable to perform further upgrades.
- other exceptions are also now handled and operations removed when the operation is failed.
- added a smoke test utilizing this exception handling, and asserting that a failed vnode upgrade can be recovered from

Commit:
Create Code Owner groups for: Sandboxing, Prackaging, DB and Crypto. (#3332)

Commit:
CORE-11492: Rename `rpcHost` to `restEndpointUrl` (#3336)

dickon committed:
CORE-10286, CORE-10394, CORE-10277: collapse a few classes into SoftCryptoService (#3164)

Have SoftCryptoService take a wrapping key store object and 2 caches, and the root wrapping key.

Eliminate the config classes that encapsulated the configuration of the crypto service layer. The only dynamic configuration was for the caches, which are now passed in. It is therefore possible to simply configure the caches in SoftCrytpoServiceProviderImpl instead.

Port all tests from the deleted classes, removing a few duplicates. Rework some of the test cases to avoid test instrumentation. Clean up the test code to reduce repetition. Share test fixtures where possible, which improves test runtime.

Inject key pair generator and wrapping key factory to the SoftCryptoService to support mocking out or replacing for testing.

Make WrappingKey an interface, so that it can be implemented for testing.

Move out the long list of acceptable key and signatures combinations into its own file.

Reduce log levels.

Rename WrappingKeyAlias.masterWrappingKey to wrappingKeyAlias for clarity.

kyriathar-r3 committed:
CORE-10556 Migrate `CryptoConnectionsFactoryImpl` from using lifecycle `AbstractComponent` (#3218)

* Migrates `CryptoConnectionsFactoryImpl` from using crypto lifecycle abstraction (`AbstractComponent`) to using plain lifecycle library

dickon committed:
CORE-6044: change health port for combined worker

Use port 7004 for combined worker since it is often run on Macs where port 7000 is not available. At this tage we won't change the separated workers since that may require significant downstream changes and separated workers are typically run in containers where there is unlikely to already something is already listening on 7000.

williamvigorr3 committed:
CORE-10678 Merge remote-tracking branch 'origin/release/os/5.0-Beta2' into WillV/CORE-10678-Merge-beta-2-into-release-os-5

williamvigorr3 committed:
CORE-10678 Merge pull request #3343 from corda/WillV/CORE-10678-Merge-beta-2-into-release-os-5

Merge release/os/5.0-Beta2 into release/os/5.0

Commit:
CORE-11492: Remove hardcoded `localhost:8888` from E2e tests (#3340)

Also remove default constructor from `TestToolkitProperty` to avoid temptation of using default REST endpoint address.

The added benefit here is that E2e tests could be executed against REST endpoint running on any host/port.
The settings will be controlled by the environment variables: `E2E_CLUSTER_B_REST_HOST` and `E2E_CLUSTER_B_REST_PORT` which can be set prior to starting E2E suite.

I have tested locally using [Ngrok](https://ngrok.com/) that it is possible to successfully run E2e using:
```
E2E_CLUSTER_B_REST_PORT=443
E2E_CLUSTER_B_REST_HOST=76ea-81-148-212-130.eu.ngrok.io
```

josephzunigadaly committed:
CORE-11434: Merge release/os/5.0-Beta2 into release/os/5.0

josephzunigadaly committed:
CORE-11434: Fix unused import from merge

josephzunigadaly committed:
CORE-11434: Build against alpha API version

filipesoliveira committed:
CORE-5630 - Replaced a constructor call with a call to the respective builder. This will allow us to update the API without causing breaking changes (#3348)

josephzunigadaly committed:
CORE-11434: Build against beta API version

josephzunigadaly committed:
CORE-11434: Merge release/os/5.0-Beta2 into release/os/5.0

Commit:
CORE-4396: Extend validity of REST self-signed certificate (#3350)

Also update instructions and minor fix to `SslCertReadServiceStubImpl`

ben-millar committed:
CORE-9664 Checking if HoldingIdentity exists in our network before attempting to establish a flow with it (#3293)

A filter was added to the `FlowGlobalPostProcessor` to detect if a counterparty exists in our network before we attempt to establish a flow with them; a timeout window has been introduced for this counterparty to materialise before the flow is errored.

nkovacsx committed:
CORE-10383 Add query registration implementation (#3327)

* CORE-10383 create custom query registration impl

* detekt

* vbump

* formatting

* Move registry to runtime

* fix test

* separate builder/factory and match with utxo

* fix compile

* fix logic and tests

* comment addressing pt1

* created collected builder impl

* vbump

* vbump

* remove dependency

chrisr3 committed:
CORE-11705: Upgrade to Corda Gradle plugins 7.0.2-SNAPSHOT.

ronanbrowne commented:
FYI the beta2 release from this repo will likely be done at the end of this coming week, will we have 7.0.2  (non-snapshot) here by then?

chrisr3 commented:
> FYI the beta2 release from this repo will likely be done at the end of this coming week, will we have 7.0.2 (non-snapshot) here by then?

Yes, planning on it. Altough Bogdan has highlighted a small addition I will need to make before then too.

Commit:
CORE-11534: Rename RPCOps to REST in `cpi-upload-rpcops-service` module (#3352)

Also renaming the module itself

nikinagy committed:
CORE-9702 - Add query for queued registration requests (#3239)

Query for queued registration requests (requests having NEW status) with a limit.

Commit:
CORE-11534: Eliminate use of `RegistrationHandle` in `CpiUploadRestResourceImpl` (#3357)

Commit:
CORE-11534: Remove references to HTTP RPC in `virtual-node-rpcops-service-impl` module (#3356)

... and rename the module.

anien committed:
CORE-9332: Align `DigitalSignatureVerificationService.verify` overloads (#3342)

- Align with the API changes in `DigitalSignatureVerificationService.verify`

- Rename `clearData` to `originalData` to align with the interface.

dimosr committed:
Merge branch 'release/os/5.0-Beta2' into dimos/os-5.0-Beta2-merge-13-03-2023

dimosr committed:
Merge pull request #3361 from corda/dimos/os-5.0-Beta2-merge-13-03-2023

CORE-11411 - Beta 2.0 to main forward merge

relyafi committed:
CORE-11299 Notary API cleanup (#3331)

Commit:
CORE-11534: Remove HTTP RPC from `virtual-node-rpcops-maintenance-impl` module (#3359)

... and rename the module.

nikinagy committed:
CORE-10698 - Fix notary validations (#3351)

Fix for notary service name validations when a member, notary joins. We should avoid having the same name for virtual nodes and notary services.

vlajos committed:
CORE-9980 send/receive transaction builder (#3195)

* Implementation of UtxoLedgerServices:
  * sendAndReceiveTransactionBuilder()
  * receiveTransactionBuilder()
  * sendUpdatedTransactionBuilder()
* The back chain resolution got updated to support working with multiple transactions in the same call.
* Disallow duplication in transaction builder.
* Remove unused TimeWindowFromImpl.

blsemo committed:
Merge commit '9421f6d' into christians/CORE-11302-merge-to-main-retry

fowlerrr committed:
CORE-11034: Add a cache per sandbox type and make the size of each configurable (#3344)

The cache is stored away in MultiCache which holds a map of one cache per SandboxGroupType. Where possible, the calling interfaces weren't changed. In some cases (see CacheControl, for example) the SandboxGroupType had to be added to distinguish between which cache a call was for. This should be external to any Sandbox users, however, as at their level it's the VirtualNodeContext which they'll (still) be passing in.

blsemo committed:
Merge pull request #3368 from corda/christians/CORE-11302-merge-to-main-retry

CORE-11302 merge to main

MrMatthewLayton committed:
CORE-10793 : Refactor isRelevant to isVisible (#3294)

Refactored `Contract.isRelevant` to `Contract.isVisible`, which aligns better with the notion of `ContractState` visibility. This now relies on `VisibilityChecker`, which provides functionality to determine state visibility. The default implementation still considers that only contract states are only visible to their participants.

Commit:
CORE-11523: `rest-stubs-impl` module refactoring (#3366)

This is a preparatory activity ahead of main work on referenced Jira. There is no new functionality introduced, but the scope of the changes is:
- Move `HelloRestResource` away from stubs module
- Rename `rest-stubs-impl` module to `ssl-cert-read-impl`
- Change `SslCertReadService` to take `SmartConfig` when creating `KeystoreInfo`
- Tidy-up dependencies, package names, etc.

kyriathar-r3 committed:
CORE-9724 Make `SecureHash` an interface on public API (#3338)

* Implements `SecureHash` and updates use sites

* Adds `parseSecureHash` static function to parse secure hashes in string form into `SecureHash` (ex public API `SecureHash.parse`)

* `DigestServiceImpl.parseSecureHash` to delegate to static `parseSecureHash` static function

dimosr committed:
CORE-11680 - Improve error message when cluster configuration is misaligned with MGM registration context

dimosr committed:
Merge pull request #3374 from corda/CORE-11680-improve-error-message

Improve error message when cluster configuration is misaligned with MGM registration context

dickon committed:
CORE-11640: start JPA repository layer, currently inert (#3346)

We should follow the JPA repository pattern in the Crypto persistence code. In this change we add the new code, not referenced yet from production code. Take care to make the new structure highly unit testable, by developing it using test driven development.

In future releases, we will need to support upgrade of the Crypto schema, which is challenging since tables are shared between cluster and virtual nodes. In production the crypto processor has many database connections, most to per-virtual-node databases and a small number to cluster databases. The timing of virtual node upgrade is up to the owner of the virtual node; we will not require everyone to upgrade at the same time. Therefore, structure the code with a "V1" in the schema name, so that we can later have "V2", "V3" etc, and potentially use any of them on a per database choice.

Stop caching the virtual node database connections. DbConnectionManager is the right place to do that, though it doesn't do so yet.

Take care over closing down the EntityManager and EntityManagerFactory when appropriate, but not for the EntityManagerFactory for the cluste which is owned by DbConnectionManager.

Commit:
CORE-11651 - Rename metrics tags to conform to the micrometer guidelines. (#3367)

* Rename metrics tags to conform to the micrometer guidelines.

Co-authored-by: David Currie <davidcurrie@users.noreply.github.com>

lankydan committed:
CORE-9595 System flow versioning (#3238)

Provides the ability to version inline system flows so that a group of 
sessions can guarantee that they are executing the same version of a 
flow pair.

System flows are versioned by platform version.

A system flow may or may not provide new versions of the flow. When a 
new version is added (versioning must be used at the start or a breaking
change must occur) that version will be used if the platform version 
agreed amongst sessions is high enough.

The platform version of a flow pair is determined by finding the lowest 
common platform version amongst a group of sessions. This platform 
version information is extracted from the MGM via the `MemberLookup` 
service.

To version a flow call:
- `VersioningService.versionedSubFlow` with a 
  `VersionedSendFlowFactory` on the initiating/starting side of the 
  flow pair.
- `VersioningService.versionedSubFlow` with a 
  `VersionedReceiveFlowFactory` on the receiving side of the flow pair.
  
The versioned flow factories construct instances of versioned flows. 
They receive a version and then the factory should have a `when` block 
that returns a `SubFlow` instance that supports that platform version.

The versioning flows are not interacted with directly. These flows 
manage the sending and receiving of version information. This includes 
the creation of `FlowSession`s that piggyback a session's first within a
versioned flow to also send version information. There is then a 
matching receiving session to pair with this logic.

When a platform version has been determined (on both sides), the version
is placed into the flow context properties. If a new versioned subFlow 
is executed when there is already a version in the context properties, 
then that version will be used and the sending of the version will be 
skipped. This allows us to simplify the code.
 
The currently versioned flows are:
- `ConsensualFinalityFlow`
- `ConesnsualReceiveFinalityFlow`
- `UtxoFinalityFlow`
- `UtxoReceiveFinalityFlow`
- `TransactionBackchainResolutionFlow`
- `TransactionBackchainSenderFlow`

YashNabar committed:
CORE-10574 Member Lookup API query by status (#3362)

Adds an extra query field to the member lookup REST endpoint so the MGM operator can query for MemberInfos by a list of statuses (ACTIVE and/or SUSPENDED).

kyriathar-r3 committed:
CORE-11521 Remove `DigitalSignature.WithKey.context` (#3347)

`DigitalSignature.WithKey.context` is returned as part of the signature. Callers of sign operation have no control over it so it's of no use to them. As such `DigitalSignature.WithKey.context` cannot be signed and is not signed. `DigitalSignature.WithKey.context` was originally added to carry untrusted (unsigned) metadata but was not really used.

- Aligns with having removed from API `DigitalSignature.WithKey.context`
